cpp/src/parquet/metadata.cc

```diff
@@ -649,9 +649,12 @@ class FileMetaData::FileMetaDataImpl {
     std::string key = file_decryptor_->GetFooterKey();
     std::string aad = encryption::CreateFooterAad(file_decryptor_->file_aad());
 
-    auto aes_encryptor = encryption::AesEncryptor::Make(
-        file_decryptor_->algorithm(), static_cast<int>(key.size()), true,
-        false /*write_length*/, nullptr);
+    auto aes_encryptor = encryption::AesEncryptor::Make(file_decryptor_->algorithm(),
+                                                        static_cast<int>(key.size()),
+                                                        true, false /*write_length*/);
+    if (ARROW_PREDICT_FALSE(aes_encryptor == nullptr)) {
+      throw ParquetException("Could not create AES encryptor for signature verification");
```

Comment:
@adamreeve @wgtmac I'm not so familiar with logic here but `AesEncryptor::Make` might return nullptr if aes not enable, so should we add this check?

Reply:
exception is better than failed and core dump here?

Reply:
Yes I think this should be checked and I agree an exception is nicer than abort.

It looks like other places that construct an AesEncryptor don't check for null either though, eg. the encryptors returned from `InternalFileEncryptor::GetMetaAesEncryptor` and `InternalFileEncryptor::GetDataAesEncryptor` appear to be used without checking for null. Maybe it would be better to change `Make` to return `Result<std::unique_ptr<AesEncryptor>>` to enforce checking for failure, or possibly it makes sense to just change the `_nossl` versions of `Make` to throw an exception rather than return null?

Reply:
> Maybe it would be better to change Make to return Result<std::unique_ptr<AesEncryptor>> to enforce checking for failure, or possibly it makes sense to just change the _nossl versions of Make to throw an exception rather than return null, as I can't see anywhere that handles a null value already?

Currently I'm not familiar with encryption enough, so I just put the checking here🤔

> AesDecryptor::Make also seems to have the same problem.

Yeah I'll update it here